api/naming/naming.go

```diff
@@ -27,3 +28,17 @@ func Task(t *api.Task) string {
 }
 
 // TODO(stevvooe): Consolidate "Hostname" style validation here.
+
+// Runtime returns the runtime name from a given spec.
+func Runtime(t api.TaskSpec) string {
+	switch r := t.GetRuntime().(type) {
+	case *api.TaskSpec_Attachment:
+		return "attachment"
+	case *api.TaskSpec_Container:
+		return "container"
+	case *api.TaskSpec_Generic:
+		return strings.ToLower(r.Generic.Kind)
+	default:
+		return "none"
```

Comment:
Fair enough - fixed it.

Could you double check the way I use the indexing?

`AllowMissing: true` plus returning `false` if `naming.Runtime` yields an error.

Also, changed the control API filters to properly handle errors. If a filter was specified, then we skip the invalid runtime. If not, then we include it.